### PR TITLE
Maintainer Filter added

### LINKED ISSUE #1035: [NG]: Add "Maintainers" filter and "Maintainer" Designation to Community Member Profile
**Current Behavior**
Community members carry a number of different roles and the responsibilities that come with that role. They are also assigned the title associated with that role. For example, much is expected of a Layer5 MeshMate. Consequently, much consideration is given to Layer5 MeshMates, including a filter to view only individuals carrying that title in the community.

Similarly, a filter should be available for individuals carrying the Maintainer role.
 
**Desired Behavior**
A view filter should existing for the layer5.io/community/member profiles carrying a "Maintainer" role. See the "MeshMates" view filter on https://meshery.io/community/meshmate/members as an example.

---
**Contributor Resources**
The `layer5` repo contains two websites. The current generation and the next-generation of the layer5.io site.

If the `layer5-ng` label is absent on this issue, then this issue pertains to the current generation of the layer5.io website, which uses Jekyll and GitHub Pages. Site content is found under the [`master` branch](https://github.com/layer5io/layer5/tree/master).
- See [`layer5` contributing instructions](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md)

If the `layer5-ng` label is present on this issue, then this issue pertains to the next-generation of the layer5.io website, which uses Gatsby, Strapi, and GitHub Pages. Site content is found under the [`layer5-ng` branch](https://github.com/layer5io/layer5/tree/layer5-ng).
- See [`layer5-ng` contributing instructions](https://github.com/layer5io/layer5/blob/layer5-ng/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by dDab-panda**
Signed-off-by: Aayush <aayush.bcode@gmail.com>

**Description**

This PR fixes #1035

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Tanuj22** [comment]
@dDab-panda great work. Are you comfortable adding a dropdown for this ?(maybe this https://github.com/fraserxu/react-dropdown) This seems to be getting crowded

**Comment by dDab-panda** [comment]
> @dDab-panda great work. Are you comfortable adding a dropdown for this ?(maybe this https://github.com/fraserxu/react-dropdown) This seems to be getting crowded

Cool, I would try to add this

**Comment by dDab-panda** [comment]
> @dDab-panda great work. Are you comfortable adding a dropdown for this ?(maybe this https://github.com/fraserxu/react-dropdown) This seems to be getting crowded

Hey @Tanuj22, let me know your reviews after checking the recent commit, I had used react-select instead of the one recommended by you

File: src/pages/community/members.js [hunk 1]
```diff
@@ -10,37 +10,43 @@ import Layout from "../../components/layout";
...
+        { label: "Active Members", value: "active" },
+        { label: "Inactive Members", value: "inactive" },
+        { label: "Maintainers", value: "maintainers" }
+    ];
 
-    const toggleInactive = () => {
-        let status = inActive;
-        setActive(false);
-        setinActive(!status);
+    const [state, setState] = React.useState({
```
Comment (Tanuj22) [inline src/pages/community/members.js:25]:
```suggestion
    const [state, setState] = useState({ 
```

Comment (Tanuj22) [reply src/pages/community/members.js:25]: Can you use `members` and `setMembers`. This is more appropriate.

File: src/sections/Members-grid/index.js [hunk 2]
```diff
@@ -1,7 +1,8 @@
-import React from "react";
+import React, { Component } from "react";
```
Comment (Tanuj22) [inline src/sections/Members-grid/index.js:1]: Remove `Component`

File: src/pages/community/members.js [hunk 3]
```diff
@@ -10,37 +10,43 @@ import Layout from "../../components/layout";
...
-        if (active) return <ActiveMembers {...props} />;
-        else if (inActive) return <InactiveMembers {...props} />;
+        let status = state.value.value;
+        if (status=="active") return <ActiveMembers {...props} />;
+        else if (status=="inactive") return <InactiveMembers {...props} />;
+        else if (status=="maintainers") return <Maintainers {...props} />;
         return <AllMembers {...props} />;
     };
-
+    const selectStyles = { menu: styles => ({ ...styles, zIndex: 999 }) };
```
Comment (Tanuj22) [inline src/pages/community/members.js:41]: Can you define all styles in the `membersGrid.style.js`?

File: package.json [hunk 4]
```diff
@@ -23,6 +23,7 @@
     "@mdx-js/mdx": "^1.6.11",
     "@mdx-js/react": "^1.6.11",
     "babel-plugin-styled-components": "^1.10.7",
+    "bootstrap": "^4.5.3",
```
Comment (Tanuj22) [inline package.json:26]: Please don't use bootstrap

**Review by Tanuj22 — CHANGES_REQUESTED** [review]
This looks nice @dDab-panda . A couple of changes needed - 

- The dropdown is giving the option to type also. Remove this. 
- The default should be set to `All Members`
- The color on the dropdown should be layer5 green instead of blue. 
- The cards on hover are underlining the name blue. Remove this.

File: src/pages/community/members.js (hunk 1 above)
Comment (dDab-panda) [reply src/pages/community/members.js:25]: Okay

**Comment by dDab-panda** [comment]
@Tanuj22 
I am a total beginner in react, so I have done couple of changes.

Things I am unable to implement
The default value-> I have wrote the code of default value as shown in documentation of react select, but its not working
The layer5 color is also having issue in styling in the css in the memberGrid.style.js (I had moved the code you asked)

The last point of 'cards on hover are underlining the name blue'. I don't see that locally, the reason unknown.
Let me know some pointers on how to fix these

**Review by Tanuj22 — APPROVED** [review]
Thanks @dDab-panda :)

**Review by Tanuj22 — APPROVED** [review]

**Pull request merged**